arrows/core/tests/test_derive_metadata.cxx

```diff
@@ -0,0 +1,127 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief test derivation of new metadata traits
+
+#include <arrows/core/derive_metadata.h>
+
+#include <vital/types/geo_point.h>
+#include <vital/types/geodesy.h>
+#include <vital/types/metadata.h>
+#include <vital/types/metadata_traits.h>
+
+#include <gtest/gtest.h>
+
+#include <iostream>
+#include <memory>
+#include <sstream>
+
+namespace kv = kwiver::vital;
+
+static kwiver::vital::metadata_traits meta_traits;
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char** argv )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
+}
+
+namespace {
+
+kwiver::vital::metadata_vector
+make_metadata()
+{
+  kv::metadata_sptr m1 = std::make_shared< kv::metadata >();
+
+  // Add the double-valued traits
+  std::map< kwiver::vital::vital_metadata_tag, double >
+  double_metadata{ { kv::VITAL_META_PLATFORM_HEADING_ANGLE, 60 },
+    { kv::VITAL_META_PLATFORM_PITCH_ANGLE, 3 },
+    { kv::VITAL_META_PLATFORM_ROLL_ANGLE, 1.2 },
+    { kv::VITAL_META_SENSOR_REL_AZ_ANGLE, 5 },
+    { kv::VITAL_META_SENSOR_REL_EL_ANGLE, 4 },
+    { kv::VITAL_META_SENSOR_REL_ROLL_ANGLE, 5 },
+    { kv::VITAL_META_PLATFORM_ROLL_ANGLE, 0.5 },
+    { kv::VITAL_META_SENSOR_VERTICAL_FOV, 30 },
+    { kv::VITAL_META_SENSOR_HORIZONTAL_FOV, 40 },
+    { kv::VITAL_META_SLANT_RANGE, 300 },
+    { kv::VITAL_META_TARGET_WIDTH, 100 } };
+
+  for( auto const& elem : double_metadata )
+  {
+    auto const& trait = meta_traits.find( elem.first );
+    m1->add( trait.create_metadata_item( elem.second ) );
+  }
+
+  // Add the geo point traits
+  auto const& sensor_loc_trait =
+    meta_traits.find( kv::VITAL_META_SENSOR_LOCATION );
+  m1->add( sensor_loc_trait.create_metadata_item(
+             kv::geo_point( kv::geo_point::geo_3d_point_t{ 0, 0, 12 },
+                            kv::SRID::lat_lon_WGS84 ) ) );
+
+  auto const& frame_center_trait =
+    meta_traits.find( kv::VITAL_META_FRAME_CENTER );
+  m1->add( frame_center_trait.create_metadata_item(
+             kv::geo_point( kv::geo_point::geo_3d_point_t{ 0, 0, 0 },
+                            kv::SRID::lat_lon_WGS84 ) ) );
+
+  // Replicate these values except without the slant range field
+  auto const m2 = std::make_shared< kv::metadata >( m1->deep_copy() );
+  m2->erase( kv::VITAL_META_SLANT_RANGE );
+  return { m1, m2 };
+}
+
+} // end namespace
+
+class derive_metadata : public ::testing::Test
+{
+  void
+  SetUp()
+  {
+    auto meta = make_metadata();
+    size_t const frame_height{ 720 };
+    size_t const frame_width{ 1080 };
```

Comment:
This is... an interesting way to declare variables?
```suggestion
    constexpr auto frame_height = size_t{ 720 };
    constexpr auto frame_width = size_t{ 1080 };
```

...although TBH these can probably just be passed to `compute_derived_metadata` directly.